src/Adapter/AbstractAdapter.php

```diff
@@ -4,5 +4,21 @@
 
 class AbstractAdapter implements AdapterInterface
 {
+    protected $trainingData;
+
+    public function getTrainingData()
+    {
+        return $this->trainingData;
+    }
+
+    public function train()
+    {
+        // TODO: Implement train() method.
+    }
+
+    public function predicte()
```

Comment:
Honestly, I don't know yet if this makes sense. Let's have a rain check on that.